MoreLinq.Test/AggregateRightTest.cs

```diff
@@ -0,0 +1,124 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MoreLinq.Test
+{
+    [TestFixture]
+    public class AggregateRightTest
+    {
+        // Overload 1 Test
+
+        [Test]
+        public void AggregateRight1WithNullSequence()
+        {
+            Assert.ThrowsArgumentNullException("source", 
+                () => MoreEnumerable.AggregateRight<int>(null, (a, b) => a + b));
+        }
+
+        [Test]
+        public void AggregateRight1WithNullFunc()
+        {
+            Assert.ThrowsArgumentNullException("func", 
+                () => Enumerable.Range(1, 5).AggregateRight(null));
+        }
+
+        [Test]
+        public void AggregateRight1WithEmptySequence()
+        {
+            Assert.Throws<InvalidOperationException>(
+                () => new int[] { }.AggregateRight((a, b) => a + b));
+        }
+
+        [Test]
+        public void AggregateRight1SimpleTest()
```

Comment:
Remove `SimpleTest` suffix from all. It serves no purpose. It's not like there's a _complicated test_. 😉 

Also the arbitrary number qualification to identify an overload should be replaced with a _symbolic_ (i.e. non-strict) concatenation of parameter names. I would recommend the following:

- `AggregateRight1` &rarr; `AggregateRight`
- `AggregateRight2` &rarr; `AggregateRightSeed`
- `AggregateRight3` &rarr; `AggregateRightResultor` or `AggregateRightResultSelector`

Have a look at [`GroupAdjacent` tests](https://github.com/morelinq/MoreLINQ/blob/07a3420b48ef2af28295cb21f794870b763cffb7/MoreLinq.Test/GroupAdjacentTest.cs) for an example of this naming where you have many overloads.